src/high-level/constructors.lisp

```diff
@@ -0,0 +1,173 @@
+;;;; constructors.lisp
+;;;;
+;;;; Author: Cole Scott
+
+(in-package #:magicl)
+
+(defconstant +default-tensor-type+ 'double-float)
+
+(defgeneric make-tensor (class shape &key initial-element layout storage)
+  (:documentation "Make a dense tensor with elements of the specified type"))
+
+(defun empty (shape &key (type +default-tensor-type+) layout)
+  "Create a tensor without intializing the contents of the storage
+
+If TYPE is not specified then +DEFAULT-TENSOR-TYPE+ is used.
+LAYOUT specifies the internal storage represenation ordering of the returned tensor.
+The tensor specialized on the specified SHAPE and TYPE."
+  (policy-cond:with-expectations (> speed safety)
```

Comment:
I think it's always beneficial to declare the type, both for speed and for safety. I'd have gone with `(declare (type shape shape))`.

Reply:
I will make a PR after this one to go back and switch to declaring the types